src/layer.js

```diff
@@ -310,8 +327,16 @@ export class MapLayer extends HTMLElement {
       let layer = this._layer,
         map = layer?._map;
       if (map) {
-        let count = 0,
-          total = 0,
+        let mapExtents = this.querySelectorAll('map-extent');
```

Comment:
This section is throwing a lot of errors in experiments, perhaps because we need to query both the light dom and shadow root for map-extent?

Reply:
I think the errors are thrown because this.querySelector returns a Nodelist, which is slightly different from array and does not have a `concat` method. Replace `concat` with the spread syntax ` ... `should work.